server/channels/store/sqlstore/session_store.go

```diff
@@ -164,6 +164,28 @@ func (me SqlSessionStore) GetSessionsWithActiveDeviceIds(userId string) ([]*mode
 	return sessions, nil
 }
 
+func (me SqlSessionStore) GetMobileVersions() ([]*model.MobileVersionMetric, error) {
+	query, args, err := me.getQueryBuilder().
+		Select(fmt.Sprintf(
+			"COUNT(userid), SPLIT_PART(deviceid, ':', 1) AS platform, COALESCE(props->>'%s','N/A') AS version, COALESCE(props->>'%s','false') as notificationDisabled",
```

Comment:
Did we check the query plan for this query? We have a lot of functions being done within SQL, as well as no filtering on the sessions table which for some customers might be really really large.

Since this is a job it likely won't be disruptive, however we should see how this performs on large data sets anyways.

Reply:
I checked in my local with 30k sessions, and I got this query plan:

```
                                                                                                QUERY PLAN                                                                                                
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 GroupAggregate  (cost=6323.64..7225.68 rows=30068 width=104)
   Group Key: (COALESCE((props ->> 'mobile_version'::text), 'N/A'::text)), (split_part((deviceid)::text, ':'::text, 1)), (COALESCE((props ->> 'device_notification_disabled'::text), 'false'::text))
   ->  Sort  (cost=6323.64..6398.81 rows=30068 width=123)
         Sort Key: (COALESCE((props ->> 'mobile_version'::text), 'N/A'::text)), (split_part((deviceid)::text, ':'::text, 1)), (COALESCE((props ->> 'device_notification_disabled'::text), 'false'::text))
         ->  Seq Scan on sessions  (cost=0.00..2134.19 rows=30068 width=123)
```

The execution of the query itself seemed immediate in my local. The results (mostly generated randomly) looked like this:
```
 count |    platform     | version | notificationdisabled 
-------+-----------------+---------+----------------------
  1343 |                 | 2.20.0  | false
   648 |                 | 2.20.0  | true
  1313 | android_rn-v2   | 2.20.0  | false
   649 | android_rn-v2   | 2.20.0  | true
  1329 | apple_rnbeta-v2 | 2.20.0  | false
   699 | apple_rnbeta-v2 | 2.20.0  | true
  1312 |                 | 2.21.0  | false
   711 |                 | 2.21.0  | true
  1326 | android_rn-v2   | 2.21.0  | false
   676 | android_rn-v2   | 2.21.0  | true
  1306 | apple_rnbeta-v2 | 2.21.0  | false
   653 | apple_rnbeta-v2 | 2.21.0  | true
  1352 |                 | 2.21.1  | false
   677 |                 | 2.21.1  | true
  1313 | android_rn-v2   | 2.21.1  | false
   662 | android_rn-v2   | 2.21.1  | true
  1378 | apple_rnbeta-v2 | 2.21.1  | false
   637 | apple_rnbeta-v2 | 2.21.1  | true
  1300 |                 | 2.22.0  | false
   668 |                 | 2.22.0  | true
  1359 | android_rn-v2   | 2.22.0  | false
   660 | android_rn-v2   | 2.22.0  | true
  1375 | apple_rnbeta-v2 | 2.22.0  | false
   639 | apple_rnbeta-v2 | 2.22.0  | true
  1949 |                 | N/A     | false
  2044 | android_rn-v2   | N/A     | false
  2023 | apple_rnbeta-v2 | N/A     | false
(27 rows)
```

Reply:
Okay looks good, we're not seeing any big "table scan" tasks which are usually the slow ones. If it's running quickly on your machine we should be okay. We can double check in load testing if need be.